pkg/jobs/jobsprotectedts/jobs_protected_ts_manager.go

```diff
@@ -99,10 +99,7 @@ func NewManager(
 // the caller may not invoke this early enough in the transaction. Returns a Cleaner
 // function to cancel installation or remove the protected  timestamp.
 func (p *Manager) TryToProtectBeforeGC(
-	ctx context.Context,
-	jobID jobspb.JobID,
-	tableDesc catalog.TableDescriptor,
-	readAsOf hlc.Timestamp,
+	ctx context.Context, job *jobs.Job, tableDesc catalog.TableDescriptor, readAsOf hlc.Timestamp,
```

Comment:
@fqazi Do schema change jobs guarantee that the in memory job record will be up to date with the persisted one? If not, i think I'll keep the current methods as is, and create a new function that allows restore to pass in `*jobs.Job`.

Reply:
They should be up to date in the schema change world as well, sorry I missed this some how. @msbutler